src/log-hook.ts

```diff
@@ -0,0 +1,224 @@
+import type { MongoDBOIDCLogEventsMap, TypedEventEmitter } from './types';
+
+/** @public */
+export interface MongoLogWriter {
+  info(c: string, id: unknown, ctx: string, msg: string, attr?: unknown): void;
+  warn(c: string, id: unknown, ctx: string, msg: string, attr?: unknown): void;
+  error(c: string, id: unknown, ctx: string, msg: string, attr?: unknown): void;
+  mongoLogId(this: void, id: number): unknown;
+}
+
+/**
+ * Connect a log event emitter instance such as the one attached to a
+ * `MongoDBOIDCPlugin` instance to a log writer that follows the format
+ * provided by the `mongodb-log-writer` npm package.
+ *
+ * @public
+ */
+export function hookLoggerToMongoLogWriter(
+  emitter: TypedEventEmitter<MongoDBOIDCLogEventsMap>,
+  log: MongoLogWriter,
+  contextPrefix: string
+): void {
+  const { mongoLogId } = log;
+
+  emitter.on('mongodb-oidc-plugin:local-redirect-accessed', (ev) => {
+    log.info(
+      'OIDC-PLUGIN',
+      mongoLogId(1_002_000_001),
+      `${contextPrefix}-oidc`,
+      'Local redirect accessed',
+      {
+        ...ev,
+      }
+    );
+  });
+
+  emitter.on('mongodb-oidc-plugin:oidc-callback-accepted', (ev) => {
+    log.info(
+      'OIDC-PLUGIN',
+      mongoLogId(1_002_000_002),
+      `${contextPrefix}-oidc`,
+      'OIDC callback accepted',
+      {
+        ...ev,
+      }
+    );
+  });
+
+  emitter.on('mongodb-oidc-plugin:oidc-callback-rejected', (ev) => {
+    log.warn(
+      'OIDC-PLUGIN',
+      mongoLogId(1_002_000_003),
+      `${contextPrefix}-oidc`,
+      'OIDC callback rejected',
+      {
+        ...ev,
+      }
+    );
+  });
+
+  emitter.on('mongodb-oidc-plugin:unknown-url-accessed', (ev) => {
+    log.info(
+      'OIDC-PLUGIN',
+      mongoLogId(1_002_000_004),
+      `${contextPrefix}-oidc`,
+      'Unknown URL accessed',
+      {
+        ...ev,
+        // strip away any query/search string (after ?) in the URL
+        path: new URL(ev.path, 'http://dummy/').pathname,
```

Comment:
Didn't know about this [base URL](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) param - cool!

![Screenshot 2023-03-14 at 2 25 39 PM](https://user-images.githubusercontent.com/1791149/225102373-5835c38f-45d5-4550-85ba-7ffce1b45c19.png)